src/compiler/checker.ts

```diff
@@ -38810,8 +38830,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function checkExpressionWithContextualType(node: Expression, contextualType: Type, inferenceContext: InferenceContext | undefined, checkMode: CheckMode): Type {
+        const filteredContextualType = filterContextualTypeForLiteralExpressionOfObject(node, contextualType);

```

Comment:
Not just `contextualType = filterContextualTypeForLiteralExpressionOfObject(node, contextualType)`?

Reply:
I can do that, no problem. I was probably just trying to avoid parameter reassignment at the time.